components/tidb_query/src/expr/builtin_json.rs

```diff
@@ -194,6 +208,41 @@ mod tests {
     use crate::expr::{EvalContext, Expression};
     use tipb::ScalarFuncSig;
 
+    #[test]
+    fn test_json_length() {
+        let cases = vec![
+            (Some("null"), None, None),
+            (Some(r#"{"a":{"a":1},"b":2}"#), Some(b"$".to_vec()), Some(2)),
+            (Some("1"), None, Some(1)),
+            (
+                Some(r#"{"a": [1, 2, {"aa": "xx"}]}"#),
+                Some(b"$.*".to_vec()),
+                None,
+            ),
+            (Some(r#"{"a":{"a":1},"b":2}"#), Some(b"$".to_vec()), Some(2)),
+        ];
+        let mut ctx = EvalContext::default();
+        for (input, param, exp) in cases {
+            let json = datum_expr(match input {
+                None => Datum::Null,
+                Some(s) => Datum::Json(s.parse().unwrap()),
+            });
+            let param = datum_expr(match param {
+                None => Datum::Null,
+                Some(b) => Datum::Bytes(b),
+            });
+
+            let op = scalar_func_expr(ScalarFuncSig::JsonLengthSig, &[json, param]);
```

Comment:
The test doesn't seem totally correct. It tests the case when the second argument is null (which shouldn't happen IMO), but missing the case when there is only one argument.

Reply:
done